xaitk_saliency/impls/perturb_image/rise.py

```diff
@@ -78,8 +81,15 @@ def __init__(
         self.threads = threads
 
         # Generate a set of random grids of small resolution
-        grid: np.ndarray = np.random.default_rng(seed).random((n, s, s)) < p1
-        grid = grid.astype('float32')
+        if type(s) is tuple:
+            if None in s:
+                grid = None  # type: Optional[np.ndarray]
+            else:
+                grid = np.random.default_rng(seed).random((n, *s)) < p1
+                grid = grid.astype('float32')
+        else:
+            grid = np.random.default_rng(seed).random((n, s, s)) < p1
+            grid = grid.astype('float32')
```

Comment:
It would be more thematic to duck-typing to attempt to *-expand `s` and catch the `TypeError` raised when failing to expand a single integer, as opposed to specifically catching the tuple type.

Reply:
I think this comment is defunct since I didn't realize a "dynamic" mode was being used in the tuple-with-a-None case.